File: BinaryPreorder.java
/**
 * Definition for a binary tree node.
 * public class TreeNode {
 *     int val;
 *     TreeNode left;
 *     TreeNode right;
 *     TreeNode() {}
 *     TreeNode(int val) { this.val = val; }
 *     TreeNode(int val, TreeNode left, TreeNode right) {
 *         this.val = val;
 *         this.left = left;
 *         this.right = right;
 *     }
 * }
 */
class Solution {
    public void preordertraversal(TreeNode root, List<Integer>order){
        if(root == null)
         return;
         order.add(root.val);
         preordertraversal(root.left, order);
         preordertraversal(root.right, order);
    }
    public List<Integer> preorderTraversal(TreeNode root) {
        List<Integer>preorder = new ArrayList<>();
        preordertraversal(root, preorder);
        return preorder;
    }
}

File: easy/BinaryPreorder.cpp
class Solution {
public:
    void preorderHelper(TreeNode* root, vector<int>& order) {
        if (root == NULL) return;

        order.push_back(root->val);
        preorderHelper(root->left, order);
        preorderHelper(root->right, order);
    }

    vector<int> preorderTraversal(TreeNode* root) {
        vector<int> preorder;
        preorderHelper(root, preorder);
        return preorder;
    }
};